Orange/widgets/evaluate/owrocanalysis.py

```diff
@@ -601,6 +604,52 @@ def _setup_plot(self):
                 warning = "All ROC curves are undefined"
         self.warning(warning)
 
+    def _on_mouse_moved(self, pos):
+        curves = [crv for crv in self.plot.curves if isinstance(crv, pg.PlotCurveItem)]
+        for i in range(len(self.selected_classifiers)):
+            sp = curves[i].childItems()[0]
```

Comment:
This assumes implicitly that the order of the curves in the `self.plot.curves` corresponds to the `self.selected_classifiers`. This seems fragile.

You can use:
```python
curves = [(clsf_idx, self.plot_curves(self.target_index, clsf_idx), self.curve_data(self.target_index, clsf_idx))
          for clsf_idx in self.selected_classifiers]  # type: List[Tuple[int, plot_curves, ROCData]]
```
to retrieve triples of (classifier_index, plot_curves, curve_data). Use `plot_curves` similarly as `curve_data` in ROC averaging dispatch below, but use

```python
plot_curves.merge().curve_item
plot_curves.avg_vertical().curve_item
plot_curves.avg_threshold().curve_item
```
to get the `pg.PlotCurveItem` instances

Reply:
My bad - should be fixed now. I've also fixed some other bugs:
- if mouse was hovered over a point on curve for some classifier and then the same classifier's curve would be hidden and mouse hovered over the same point as before, the cache would not reset and therefore an invalid threshold would be shown,
- if tooltip was not being shown and the cache point was "valid" (i. e. same point as previously shown), a tooltip with message "" would be created, which resulted in hiding a tooltip.

I have also encountered the following exception (just once), which I can not seem to be able to reproduce anymore:
```
--------------------------- RuntimeError Exception ----------------------------
Traceback (most recent call last):
  File "/home/matej/.local/lib/python3.5/site-packages/pyqtgraph/GraphicsScene/GraphicsScene.py", line 162, in mouseMoveEvent
    self.sendHoverEvents(ev)
  File "/home/matej/.local/lib/python3.5/site-packages/pyqtgraph/GraphicsScene/GraphicsScene.py", line 228, in sendHoverEvents
    items = self.itemsNearEvent(event, hoverable=True)
  File "/home/matej/.local/lib/python3.5/site-packages/pyqtgraph/GraphicsScene/GraphicsScene.py", line 424, in itemsNearEvent
    if hoverable and not hasattr(item, 'hoverEvent'):
RuntimeError: wrapped C/C++ object of type ScatterPlotItem has been deleted
```

When browsing through old issues, I've found out that there's been some instances of similar issues (#1316), though I can't say I completely understand your fix for this.